common-service-core/src/main/java/org/zowe/apiml/security/HttpsFactory.java

```diff
@@ -241,11 +237,13 @@ private void setSystemProperty(String key, String value) {
 
     public void setSystemSslProperties() {
         setSystemProperty("javax.net.ssl.keyStore", SecurityUtils.replaceFourSlashes(config.getKeyStore()));
-        setSystemProperty("javax.net.ssl.keyStorePassword", config.getKeyStorePassword());
+        setSystemProperty("javax.net.ssl.keyStorePassword",
+            config.getKeyStorePassword() == null ? null : String.valueOf(config.getKeyStorePassword()));
```

Comment:
Isn't this defeating the purpose of this PR?

Reply:
Yes, this is problematic part of this issue, everywhere where is a password stored in String is problematic (including authorization header in server request, etc.).

It will be great from point of view security, remove this system property at all. But the purpose of this issue is just mitigated using String. My aim in this issue is a shared library to API SDK.